lisa/tools/service.py

```diff
@@ -72,9 +77,11 @@ def stop_service(self, name: str) -> None:
             cmd_result = self.run(f"{name} stop", shell=True, sudo=True, force_run=True)
             cmd_result.assert_exit_code()
 
-    def restart_service(self, name: str) -> None:
+    def restart_service(self, name: str, ignore_error_code: int = 0) -> None:
         cmd_result = self.run(f"{name} restart", shell=True, sudo=True, force_run=True)
-        cmd_result.assert_exit_code()
+        # optionally ignore exit code if it matches our expected non-zero value
+        if not (ignore_error_code and cmd_result.exit_code == ignore_error_code):
```

Comment:
How about we skip `assert_exit_code` check and do something like `assert_that(cmd_result.exit_code, message).is_in(0, ignore_error_code)`? We could customize the message to say that exit code was not in the list of expected value